packages/ledger/src/index.ts

```diff
@@ -0,0 +1,270 @@
+import { accountSelect, createEIP1193Provider } from '@bn-onboard/common'
+import {
+  ScanAccountsOptions,
+  Account,
+  Asset
+} from '@bn-onboard/common/src/types'
+import type { Chain, WalletInit } from '@bn-onboard/types'
+import type { BIP32Interface } from 'bip32'
+import type Transport from '@ledgerhq/hw-transport'
+
+const LEDGER_LIVE_PATH = `m/44'/60'`
+const LEDGER_DEFAULT_PATH = `m/44'/60'/0'`
+
+const DEFAULT_BASE_PATHS = [
+  {
+    label: 'Ledger Live',
+    value: LEDGER_LIVE_PATH
+  },
+  {
+    label: 'Ethereum',
+    value: LEDGER_DEFAULT_PATH
+  }
+]
+
+const assets = [
+  {
+    label: 'ETH'
+  }
+]
+
+const getFullDerivationPath = (derivationPath: string, index: number): string =>
+  derivationPath === LEDGER_LIVE_PATH
+    ? `${derivationPath}/${index}'/0/0`
+    : `${derivationPath}/${index}`
+
+type CustomNavigator = Navigator & { usb: { getDevices(): void } }
+
+const supportsWebUSB = (): Promise<boolean> =>
+  Promise.resolve(
+    !!navigator &&
+      !!(navigator as CustomNavigator).usb &&
+      typeof (navigator as CustomNavigator).usb.getDevices === 'function'
+  )
+
+/**
+ * Returns the correct ledger transport based on browser compatibility for webUSB.
+ * @returns
+ */
+const getTransport = async () =>
+  ((await supportsWebUSB())
+    ? (await import('@ledgerhq/hw-transport-webusb')).default
+    : (await import('@ledgerhq/hw-transport-u2f')).default
+  ).create()
+
+interface LedgerAccount {
+  publicKey: string
+  derivationPath: string
+  chainCode: string
+}
+
+const getBalance = async (
+  address: string,
+  rpcUrl: string,
+  block: string | number = 'latest'
+): Promise<string> => {
+  const { providers } = await import('ethers')
+  const provider = new providers.JsonRpcProvider(rpcUrl)
+  return (await provider.getBalance(address, block)).toHexString()
```

Comment:
@aaronbarnardsound Balance in hex or string number? TS Types currently are string. IMO it should stay bignumber until it gets displayed so that it stays flexible up until it is time to be displayed